src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -220,6 +226,16 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             catch (ConsensusErrorException ex)
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
+
+                if (ruleContext.ValidationContext.Error != ConsensusErrors.BadTransactionDuplicate)
```

Comment:
I'm not sure if this is the best Place to have this codeIt feels very hacky here

Reply:
Feel free to suggest another place, I don't mind. 

I thought about following places as well: 
in block validator- that will require copy-pasting this code in 3 places or moving to a separated class
in a new rule- that will require making public property on consensus rule engine to expose invalid block hashes store so the rule can access it- felt like an ugly hack to me.